File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int countStair(vector<int>&v, int n){
        if(n==1 || n ==2) return n;
        if(v[n] != -1 ) return v[n];
        v[n] = countStair(v,n-1) + countStair(v,n-2);

        return v[n];
    }
    int climbStairs(int n) {
        vector<int> v(n+1,-1);
       return countStair(v,n);
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        // n=1 , 1 , for n=2 -> 2, for n=3-> 3, for n = 4 
        // 2+2 
        // 1+2+1
        // 1+1+1+1
        // 1 + 1 + 2
        // 2+ 1+ 1

        if(n==1 || n==2) return n; 
        
        int[] arr = new int[n+1];

        arr[1] = 1; 
        arr[2] = 2; 


        for(int i=3; i<=n;i++){
            arr[i] = arr[i-1] + arr[i-2];
        }

        return arr[n];
    }
}